pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -613,6 +687,16 @@ private boolean isPeerURL(String segmentLocation) {
         .startsWith(CommonConstants.Segment.PEER_SEGMENT_DOWNLOAD_SCHEME);
   }
 
+  /**
+   * Cache the LLC segment without deep store download uri to
```

Comment:
This class is getting too big. Is it possible to move some of the new code added to a new class? Will make things more readable

Reply:
Added a TODO in the class.